include/records.hrl

```diff
@@ -40,7 +40,7 @@
              msg :: binary()}).
 %% Error packet, commonly used in the protocol.
 -record(error, {code :: integer(), state :: binary() | undefined,
-                msg :: binary()}).
+                msg :: binary() | {error, term()}}).
```

Comment:
So the new error comes out as code =  -1, state = undefined, msg = `{error, not allowed}`. Not very clear. (What's not allowed?) Also the tuple might break existing user code doing things like logging the errors and assuming msg is binary. I think msg should be a descriptive sentence as for the server errors. Idea: "Server requested a file not permitted by the client: /etc/passwd". Possibly truncating the file name if > 100 chars or so. The code -1 is fine and undefined state s not new so also fine.

Reply:
Ok, changed as requested. There are now 2 error codes, though, `-1` for not allowed, `-2` for error opening file (more on that below).